File: app/src/main/java/org/alkaline/taskbrain/dsl/runtime/NoteMutation.kt
package org.alkaline.taskbrain.dsl.runtime

import org.alkaline.taskbrain.data.Note

/**
 * Represents a mutation that occurred during directive execution.
 * Used to propagate changes back to the ViewModel for cache and UI updates.
 */
data class NoteMutation(
    val noteId: String,
    val updatedNote: Note,
    val mutationType: MutationType
)

/**
 * Types of mutations that can occur during directive execution.
 */
enum class MutationType {
    /** Note path was changed */
    PATH_CHANGED,
    /** Note content was changed (includes name changes) */
    CONTENT_CHANGED,
    /** Content was appended to the note */
    CONTENT_APPENDED
}

File: web/src/dsl/runtime/NoteMutation.ts
import type { Note } from '@/data/Note'

export enum MutationType {
  PATH_CHANGED = 'PATH_CHANGED',
  CONTENT_CHANGED = 'CONTENT_CHANGED',
  CONTENT_APPENDED = 'CONTENT_APPENDED',
}

export interface NoteMutation {
  noteId: string
  updatedNote: Note
  mutationType: MutationType
  appendedText?: string
}
